### PR TITLE
Try to generate better playground code

### PR DESCRIPTION
**Pull request by noxer**
Sorry for the format change.

This PR adds the functionality to identify code sections in a post and generate a useful code for the playground. The non-code parts are inserted as comments.

I've not used the `strings.Builder` for backwards compatibility reasons. I can add it if required.

Happy Hacktoberfest,
Tim

### CODE REVIEW

**Comment by noxer** [comment]
Fixes #38

File: bot/bot.go [hunk 1]
```diff
@@ -837,6 +821,36 @@ func flipCoin(ctx context.Context, b *Bot, event *slack.MessageEvent) {
 	}
 }
 
+// postToPlayground converts the text of a post into code for the playground. It is not perfect but works most of the time.
+// Text outside of ``` quotes is converted into a comment and included in the code, everything inside of those quotes is
+// considered code and pasted as-is.
+func postToPlayground(text string) string {
+	var pg string
+	parts := strings.Split(text, "```")
```
Comment (theckman) [inline bot/bot.go]: Should we just bail if this is `< 2`?

Comment (noxer) [reply bot/bot.go]: Not sure, I've seen a few posts where the author forgot the second ```.

Comment (theckman) [reply bot/bot.go]: True, good point. 👍

**Comment by dlsniper** [comment]
Sorry, this won't be merged as the intention is to get people to use the playground when appropriate. Thank you for your time and help.

**Comment by theckman** [comment]
@dlsniper I'm not sure I follow. This will still encourage people to use the playground, but would generate more functional code when the snippet is pasted to the playground.

As someone who spends a lot of his time helping on Slack via a mobile phone, this is a common pain point for me.

Would you be able to share more context around your concerns?

**Comment by dlsniper** [comment]
@theckman we discussed this extensively to the point that I should write a bot command to reply to this.

The purpose of this is not to make the bot smarter, but the humans. The fact that humans paste large swats of code/text freely and break conversations is more annoying than a message telling the user to stop doing it.

So if you are annoyed by a single line message, I find it surprising that messages larger than 10 lines don't annoy you.

**Comment by theckman** [comment]
@dlsniper sure, I agree on the intent. But when it happens, it doesn't make sense that it generates broken code if someone has text with the code snippet.

If you're not in favor of merging this, would you be amenable to a PR just removing the functionality and put in place an alias?

**Pull request closed without merging**